kvs/client/python/anna/ipc_client.py

```diff
@@ -101,55 +99,50 @@ def get(self, refs):
 
                     # for set lattice, just pick the first one to return for now
                     # we are not using it anyway..
-                    if deserialize[tp.key]:
-                        kv_pairs[tp.key] = deserialize_val(res[0])
-                    else:
-                        kv_pairs[tp.key] = res[0]
+                    kv_pairs[tp.key] = res[0]
                 else:
-                    raise ValueError('Invalid Lattice type: ' + str(tp.lattice_type))
+                    raise ValueError('Invalid Lattice type: ' +
+                                     str(tp.lattice_type))
             return kv_pairs
 
-    def causal_get(self, refs, future_read_set, address_to_versioned_key_list_map, consistency, client_id):
+    def causal_get(self, refs, future_read_set, 
+                   versioned_key_locations, consistency, client_id):
         request = CausalRequest()
 
-        if consistency == SINGLE_OBJ_CAUSAL:
+        if consistency == SINGLE:
             request.consistency = SINGLE
-        elif consistency == CROSS_OBJ_CAUSAL:
+        elif consistency == CROSS:
             request.consistency = CROSS
         else:
-            print("Error: non causal consistency in causal mode!")
+            logging.error("Error: non causal consistency in causal mode!")
             return None
 
         request.id = str(client_id)
 
-        request.address_to_versioned_key_list_map = address_to_versioned_key_list_map
-
-        deserialize = {}
+        request.versioned_key_locations = versioned_key_locations
```

Comment:
I don't remember exactly what the Python protobuf syntax is, but I don't think this is going to work. I'm pretty sure you can't simply assign a Python map to a protobuf map. You might be able to use `CopyFrom`.